internal/test/e2e/cluster_test.go

```diff
@@ -1769,12 +1770,31 @@ func TestUploadPacking(t *testing.T) {
 	}
 
 	// upload 2 more files which are half a slab each to test filling up a slab
-	// exactly.
+	// exactly, before uploading file 5 we wait until all buffers are
+	// complete/unlocked and there are no more packed slabs left
 	data4 := make([]byte, slabSize/2)
-	data5 := make([]byte, slabSize/2)
 	uploadDownload("file4", data4)
-	uploadDownload("file5", data5)
 	download("file4", data4, 0, int64(len(data4)))
+	tt.Retry(100, 100*time.Millisecond, func() error {
```

Comment:
This is necessary to avoid an NDF in `TestUploadPacking`. It's non-deterministic because packed slabs are uploaded in a background thread asynchronously. There's a race where there's a packed slab but no background thread to upload it. In the wild that's not really an issue it'll get uploaded on the next upload but in this test we assert the object statistics causing it to fail.

Reply:
Seems like it is still failing https://github.com/SiaFoundation/renterd/actions/runs/11496910147/job/31999500762?pr=1606

Reply:
Yeah I reverted the retry because that passed the test 1000 times in a loop. Would you rather alter the upload loop? The race is pretty obvious, we check for packed slabs and break out of the "launch loop" when there's none left, and then we wait until all of them are uploaded and then we exit the background loop. In the mean time however there might've been new ones. We could change that but I'm not sure if it's worth it only to make this test pass.